python/fastexcel/__init__.py

```diff
@@ -27,6 +29,9 @@
 )
 from ._fastexcel import read_excel as _read_excel
 
+DType = Literal["null", "int", "float", "string", "boolean", "datetime", "date", "duration"]
+DTypeMap: TypeAlias = "dict[str, DType] | dict[int, DType]"
```

Comment:
It doesn't seem to work with my tests
```
ipdb> excel_file.load_sheet_by_idx(1, dtypes={"__UNNAMED__1": "boolean"}).to_polars()
shape: (24, 12)
┌────────────────────────┬──────────────┬──────────────────────────────┬──────────────┬───┬──────────────┬──────────────┬───────────────┬───────┐
│ Phone Economics Model  ┆ __UNNAMED__1 ┆ Asset Type:                  ┆ __UNNAMED__3 ┆ … ┆ __UNNAMED__8 ┆ __UNNAMED__9 ┆ __UNNAMED__10 ┆ ALL   │
│ ---                    ┆ ---          ┆ ---                          ┆ ---          ┆   ┆ ---          ┆ ---          ┆ ---           ┆ ---   │
│ str                    ┆ bool         ┆ str                          ┆ str          ┆   ┆ null         ┆ null         ┆ null          ┆ str   │
╞════════════════════════╪══════════════╪══════════════════════════════╪══════════════╪═══╪══════════════╪══════════════╪═══════════════╪═══════╡
│ null                   ┆ null         ┆ Forecast Error (less than):  ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ COO   │
│ null                   ┆ null         ┆ Production Time (Between):   ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ DELCO │
│ null                   ┆ null         ┆ Store Call Volume (Between): ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ RBD   │
│ null                   ┆ null         ┆ Dine-In Volume (Less than):  ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ RR    │
│ null                   ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ …                      ┆ …            ┆ …                            ┆ …            ┆ … ┆ …            ┆ …            ┆ …             ┆ …     │
│ All calls Instore      ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Busy Call Forward      ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Call Forward No Answer ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Promise Time           ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ 100% Call Center       ┆ null         ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
└────────────────────────┴──────────────┴──────────────────────────────┴──────────────┴───┴──────────────┴──────────────┴───────────────┴───────┘
ipdb> excel_file.load_sheet_by_idx(1, dtypes={1: "boolean"}).to_polars()
shape: (24, 12)
┌────────────────────────┬───────────────────────────────────┬──────────────────────────────┬──────────────┬───┬──────────────┬──────────────┬───────────────┬───────┐
│ Phone Economics Model  ┆ __UNNAMED__1                      ┆ Asset Type:                  ┆ __UNNAMED__3 ┆ … ┆ __UNNAMED__8 ┆ __UNNAMED__9 ┆ __UNNAMED__10 ┆ ALL   │
│ ---                    ┆ ---                               ┆ ---                          ┆ ---          ┆   ┆ ---          ┆ ---          ┆ ---           ┆ ---   │
│ str                    ┆ str                               ┆ str                          ┆ str          ┆   ┆ null         ┆ null         ┆ null          ┆ str   │
╞════════════════════════╪═══════════════════════════════════╪══════════════════════════════╪══════════════╪═══╪══════════════╪══════════════╪═══════════════╪═══════╡
│ null                   ┆ null                              ┆ Forecast Error (less than):  ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ COO   │
│ null                   ┆ null                              ┆ Production Time (Between):   ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ DELCO │
│ null                   ┆ null                              ┆ Store Call Volume (Between): ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ RBD   │
│ null                   ┆ null                              ┆ Dine-In Volume (Less than):  ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ RR    │
│ null                   ┆ null                              ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ …                      ┆ …                                 ┆ …                            ┆ …            ┆ … ┆ …            ┆ …            ┆ …             ┆ …     │
│ All calls Instore      ┆ Store answers all calls (no call… ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Busy Call Forward      ┆ Calls are forwarded to the call … ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Call Forward No Answer ┆ Calls are forwarded to the call … ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ Promise Time           ┆ All calls are forwarded after th… ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
│ 100% Call Center       ┆ Call center answers all calls (M… ┆ null                         ┆ null         ┆ … ┆ null         ┆ null         ┆ null          ┆ null  │
└────────────────────────┴───────────────────────────────────┴──────────────────────────────┴──────────────┴───┴──────────────┴──────────────┴───────────────┴───────┘
```

+ I reckon we could support `dict[str | int, Dtype]` for unnamed columns (instead of recalling to write `__UNNAMED__1`. ..)